### PR TITLE
rmg: add AGENTS.md

### PR DESCRIPTION
**Pull request by starius**
Added AGENTS.md in `rmg` package to have a place where it is described in depth. The description works for both humans and agents.

Also added AGENTS.md into modificators/ subdir.

### CODE REVIEW

**Comment by IvanSavenko** [comment]
@DjWarmonger any thoughts?

**Comment by DjWarmonger** [comment]
First of all: why rmg specifically? Do we have any other AGENTS.md already?

**Comment by starius** [comment]
Adding AGENTS.md in a subdir makes sense, because it is treated as instructions specific to that dir.

My main focus is rmg right now. I can add AGENTS.md files to all other components, give me the green light :)

**Comment by DjWarmonger** [comment]
Yeah sure, I already have nice top-level description. Though it's CLAUDE.md, I also use cursor/rules, but didn't share any othe them. Which AI tool do you use?

File: lib/rmg/AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+
+### Option Finalization
+
+`CMapGenOptions::finalize()`:
+
+ - chooses random template if missing (`getPossibleTemplate`).
+ - resolves random player/team/water/monster settings.
+ - enforces template compatibility (size/player/water constraints).
+
+## Zone Placement Algorithms
```
Comment (DjWarmonger) [inline lib/rmg/AGENTS.md]: This is part specific to CZonePlacer.cpp and irrelevant to any other stage.

Also, placement algorithm already covered in docs: https://github.com/vcmi/vcmi/blob/develop/docs/developers/RMG_Description.md
Maybe it'd bet better just to link to existing document.

File: lib/rmg/AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+ - runs jobs when `isReady()` (all dependencies finished).
+ - supports deterministic single-thread mode (`config.singleThread`) and parallel mode (TBB).
+
+Dependency helpers (macros):
+
+ - `DEPENDENCY(X)`, `POSTFUNCTION(X)`, and `_ALL` variants.
+
+Important: dependency graph is distributed across many `init()` methods; ordering bugs usually come from missing dependency edges.
+
+## Modificators: Responsibilities And Order Constraints
```
Comment (DjWarmonger) [inline lib/rmg/AGENTS.md]: Now modificators are better handled in separate AGENTS.md in their specific subfolder. No need to enumerate all of them on top level.

File: lib/rmg/AGENTS.md [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+There are explicit `FIXME/TODO` markers in core paths, notably:
+
+ - lock ordering / potential deadlocks (`WaterProxy`, cross-zone access).
+ - assumptions about player indexing in some team/owner paths.
+ - several custom object and required object serialization edge cases.
+ - template validation currently minimal (`CRmgTemplate::validate` mostly placeholder).
+
+When changing these areas, prefer incremental changes and test with both:
+
+ - single-thread mode
```
Comment (DjWarmonger) [inline lib/rmg/AGENTS.md]: Nope. Single-threaded mode is unused and could as well be removed altogether.

File: lib/rmg/AGENTS.md [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+ - multi-thread mode
+ - maps with and without water
+ - single-level and underground templates
+
+## File Map (Quick Navigation)
+
```
Comment (DjWarmonger) [inline lib/rmg/AGENTS.md]: Agent can run `ls` on its own. Also, it will learn nothing from this list. No point to clutter AGENTS.md with list of files.

**Review by DjWarmonger — CHANGES_REQUESTED** [review]

**Comment by starius** [comment]
> Yeah sure, I already have nice top-level description. Though it's CLAUDE.md, I also use cursor/rules, but didn't share any othe them. Which AI tool do you use?

I use codex and sometimes claude.

File: lib/rmg/AGENTS.md (hunk 1 above)
Comment (starius) [reply lib/rmg/AGENTS.md]: I re-wrote the file, removing all things, already explained in existing docs. They were replaced with references.

File: lib/rmg/AGENTS.md (hunk 2 above)
Comment (starius) [reply lib/rmg/AGENTS.md]: Fixed

File: lib/rmg/AGENTS.md (hunk 3 above)
Comment (starius) [reply lib/rmg/AGENTS.md]: Fixed

File: lib/rmg/AGENTS.md (hunk 4 above)
Comment (starius) [reply lib/rmg/AGENTS.md]: Fixed

**Review by DjWarmonger — APPROVED** [review]
